### DISCUSSION TITLE
How to aproach existing db

### DISCUSSION THREAD

**Discussion by francoganga** [gobuffalo/buffalo#2321] [Q&A] [answered]
What is the best way of adding something like a pgx connection for querying a existing postgres database?.

**Comment by sio4** [comment, marked as answer]
> I want to use it with the --api flag to do a simple rest api querying a existing postgres db.

OK, what I understood regarding your main point is that you don't want to create/manage new database tables using Buffalo's database management features such as migration but just want to use your existing database tables as a data store for the models of your app. Is it correct? It could be simple. Let's see the typical workflow of developing an application with Buffalo briefly.

1. scaffold the app using `buffalo new app ....`, and go into the app's directory.
2. configure DBMS connection by editing `database.yml` with the access parameters such as user, password, host, port, etc.
3. create resources by running `buffalo generate resources people name:string age:int`. Buffalo will generate the corresponding action, route, model, and migration files.
4. at this point, you may run `buffalo pop create` and `buffalo pop migrate` to create the database for the app and tables for models.
5. once database configuration is done, you can run `buffalo dev` to see if things are working.
6. if you can see the basic actions, you may start to write your own code to handle your data.

These are typical steps but in your situation, the database and tables already exist so you don't want to go through step 4. So just skip that step. That's totally fine. Some more things to be considered are:

* you can still use `buffalo g resources` to scaffold your action and models, but you don't need a migration file which is only required for the database table creation. in that case, you can run `buffalo g resource people --skip-migration`.
* since your tables are not managed by Buffalo's migration, you need to sync columns in your table and your model by editing the `models/person.go` file's `type Person struct {}` part. (the model, actually the struct Person, represents your database table in the app)

That's all!

However, if you are not familiar with Buffalo's behavior yet, starting with the pure Buffalo app as a learning material could be the best solution. Once you understand how the Buffalo app interacts with the database, developing an app with an existing database is not a hard thing.

I think this document is a good starting point to an understanding of the model.

https://gobuffalo.io/documentation/database/models/


--- original content of this comment ---
Could you explain some more about your goal with context/background or example scenario?

**Reply by francoganga** [reply]
That's it! just what i needed thank you.

**Comment by francoganga** [comment]
I want to use it with the --api flag to do a simple rest api querying a existing postgres db. And a the same time learn the language. Im aware that i probably dont need to use buffalo but i want to try it anyways.

**Reply by sio4** [reply]
I edited my previous answer (to make it easy to find from the thread). Please let me know if you are still not clear with the answer.
